ecoscope/io/async_earthranger.py

```diff
@@ -0,0 +1,719 @@
+import json
+import geopandas as gpd
+import pandas as pd
+import asyncio
+from dateutil import parser
+from erclient.client import AsyncERClient
+from ecoscope.io.utils import to_hex
+
+import ecoscope
+
+
+class AsyncEarthRangerIO(AsyncERClient):
+    def __init__(self, sub_page_size=4000, tcp_limit=5, **kwargs):
+        if "server" in kwargs:
+            server = kwargs.pop("server")
+            kwargs["service_root"] = f"{server}/api/v1.0"
+            kwargs["token_url"] = f"{server}/oauth2/token"
+
+        self.sub_page_size = sub_page_size
+        self.tcp_limit = tcp_limit
+        self.event_type_display_values = None
+
+        kwargs["client_id"] = kwargs.get("client_id", "das_web_client")
+        super().__init__(**kwargs)
+
+    @staticmethod
+    def _clean_kwargs(addl_kwargs={}, **kwargs):
+        for k in addl_kwargs.keys():
+            print(f"Warning: {k} is a non-standard parameter. Results may be unexpected.")
+        return {k: v for k, v in {**addl_kwargs, **kwargs}.items() if v is not None}
+
+    @staticmethod
+    def _normalize_column(df, col):
+        print(col)
+        for k, v in pd.json_normalize(df.pop(col), sep="__").add_prefix(f"{col}__").items():
+            df[k] = v.values
+
+    @staticmethod
+    def _dataframe_to_dict(events):
+        if isinstance(events, gpd.GeoDataFrame):
+            events["location"] = pd.DataFrame({"longitude": events.geometry.x, "latitude": events.geometry.y}).to_dict(
+                "records"
+            )
+            del events["geometry"]
```

Comment:
Can we reuse these methods in `EarthRangerIO`?

Reply:
I don't think DRY is a good enough reason alone to have an instance of the synchronous client within the async client.

A solution to this might be to have a single `earthranger` class/mixin that is wraped into both versions of the client under the hood. The class would contain synchronous utility functions like the above and probably handle all of the json -> dataframe / adding correlating keys logic that's currently repeated between the two clients.

Reply:
I've moved the `@staticmethod`s into a new module called `earthranger_utils` which is used by both clients.

Reply:
This can also be where the json->dataframe functions end up in the glorious future